components/proxy/src/main/java/com/hotels/styx/startup/ProxyServerSetUp.java

```diff
@@ -39,15 +43,34 @@ public ProxyServerSetUp(PipelineFactory pipelineFactory) {
         this.pipelineFactory = requireNonNull(pipelineFactory);
     }
 
-    public HttpServer createProxyServer(StyxServerComponents config) {
-        HttpHandler pipeline = pipelineFactory.create(config);
+    public HttpServer createProxyServer(StyxServerComponents components) {
+        // TODO see https://github.com/HotelsDotCom/styx/issues/382
+
+        while (!arePluginsLoaded(components)) {
```

Comment:
Shouldn't we put a (configurable) max time limit for the loading of plugins/startup of the app so we won't block  here forever? This could be some limit in styxServer.startAsync().awaitRunning(xxx) which is unlimited now.

Reply:
That's a good idea, though perhaps it should be a separate issue & feature-branch?
It is already possible for plugins to block Styx start-up in the `master`.

Reply:
I would include it in this PR:
In a scenario such as Kubernetes where the 'ping' endpoint is used for liveness checks, before this PR Styxpres would fail liveness checks if it got stuck and would be restarted (and that might perhaps solve the issue). However, after this PR Styxpres would be considered alive and would not be restarted.